include/service/entities/site_query_service.js

```diff
@@ -259,19 +259,68 @@ module.exports = function SiteQueryServiceModule(pb) {
    * @param {Function} cb
    */
   SiteQueryService.prototype.getCollections = function (cb) {
-      this.listCollections({}, function(err, items) {
-          if(pb.util.isError(err)) {
-            pb.log.error(err);
-            return cb(err);
-          }
+    this.listCollections({}, function(err, items) {
+      if(pb.util.isError(err)) {
+        pb.log.error(err);
+        return cb(err);
+      }
 
-          items = items.filter(function(item) {
-              return item.name.indexOf('system.indexes') === -1 && item.name.indexOf('system.namespaces') === -1;
-          });
+      items = items.filter(function(item) {
+        return item.name.indexOf('system.indexes') === -1 && item.name.indexOf('system.namespaces') === -1;
+      });
+
+      cb(err, items);
+    });
+  };
+
+  /**
+   * Deletes all site specific content
+   * @param {Array} collections -  array of collection names
+   * @param {String} siteid - unique site id
+   * @param {Function} callback - callback function
+   */
+  SiteQueryService.prototype.deleteSiteSpecificContent = function (collections, siteid, callback) {
+    var self = this;
+    var BaseObjectService = pb.BaseObjectService;
+
+    SiteService.siteExists(siteid, function (err, exists) {
+      if (util.isError(err)) {
+        return callback(err);
+      }
+
+      if (!exists) {
+        return callback(BaseObjectService.validationError([BaseObjectService.validationFailure("siteid", "Invalid siteid")]));
+      }
 
-          cb(err, items);
+      var tasks = util.getTasks(collections, function (collections, i) {
+        return function (taskCallback) {
+          self.delete({site: siteid}, collections[i].name, function (err, commandResult) {
+            if (util.isError(err) || !commandResult) {
+              return taskCallback(err);
+            }
```

Comment:
you will want a different else so you can catch the case where there is no error but the command result is null.  I don't think this is possible TBH but not positive.  I would just remove the ```|| !commandResult``` and call it a day.